### ISSUE TITLE
Compile without MS Visual Studio?

### ISSUE THREAD

**Issue by shigutso** [fireice-uk/xmr-stak#1944]
# Compile issues
  - Windows 8.1
  ```
C:\xmr-stak-master\build>cmake --build . --config Release --target install
CMake Error:
  Generator

    Visual Studio 15 2017 Win64

  could not find any instance of Visual Studio.
  ```

Is it possible to compile xmr-stak on Windows without Visual Studio? I can't install MS Visual Studio due to licensing issues where I work. Even the Community Edition is not free for companies.

**Comment by Spudz76** [comment]
No.  Also strange, Appveyor runs Community Edition in their test instances, they are a company.

Pretty sure it's licensed to whoever is using it not the computer or the company that owns said computer (like most other M$ software)

**Comment by shigutso** [comment]
There are some cases where the CE can be used in a company, but there are exceptions:
Took from https://visualstudio.microsoft.com/vs/community/
> For organizations
> An unlimited number of users within an organization can use Visual Studio Community for the following scenarios: in a classroom learning environment, for academic research, or for contributing to open source projects.
> For all other usage scenarios:
> In non-enterprise organizations, up to five users can use Visual Studio Community. In enterprise organizations (meaning those with >250 PCs or >$1 Million US Dollars in annual revenue), no use is permitted beyond the open source, academic research, and classroom learning environment scenarios described above.

Maybe it's possible to compile with MSYS?

**Comment by Spudz76** [comment]
This is an open source project.
Thus, you are totally clear to use Community Edition.  Nobody is going to care or even know.  As long as you never use the actual IDE you don't even need a Microsoft ID or license whatsoever.

You can install the build tools alone which are I believe totally free (no IDE or cool stuff, just the compilers and toolkit/headers)  This is the way Node.js NPM handles it, you run `npm install windows-build-tools` and it does it all for you regardless who you are or where you work.  However if you need older runtime/compiler combos it can be insufficent (14.0 for CUDA8, but most everything newer than that can just use the default current stuff)

I have not tried the "just build tools" angle I only suspect it might work.

MSYS or mingw or cygwin will not compile as far as I know.  Most of the code checks for MSVC defines not for more generic "is this windows?" defines.  So it would assume since not MSVC must be UNIX/GCC or compatible.  And that's just the CMake code there could be many many other places in the real C++ code where it only checks for MSVC to decide if Windows or not.

Making it work would take a full hunt for bad logic and convert all "MSVC" specific definitions to whatever the proper way to detect WIN64 (most WIN64 compilers also define WIN32 but that feels almost as wrong since this code does not do 32bit -- some alternative compilers such as gcc from cygwin64 *also* define UNIX or POSIX things, so then it gets super confusing (since it's a UNIX-4-Windows kind of situation).

Also I don't think all the deps are available for those compilers (libhwloc / libmicrohttpd) so you would be compiling those from source (maybe) and then trying to link to them etc.
